### DISCUSSION TITLE
Functions from a GatsbyJS Theme

### DISCUSSION THREAD

**Discussion by kalebheitzman** [gatsbyjs/gatsby#32237] [Help] [answered]
Is there a plugin similar to gatsby-plugin-page-creator that will map functions defined in a theme to the host Gatsby site?

When I test the function from the host site under ```src/api``` everything works great. If I move the function to the theme, Gastby doesn't register it.

Thanks!

**Comment by KyleAMathews** [comment, marked as answer]
You need to add the plugin to `my-plugin/src/api/my-plugin/my-function.js` and then you can access the function at `/api/my-plugin/my-function`

So odds are you're not putting your function in the right folder in your theme. See the PR adding this functionality https://github.com/gatsbyjs/gatsby/pull/31466

**Reply by KyleAMathews** [reply]
We should add a warning when Gatsby sees a function in a plugin/theme that isn't namespaced.

**Reply by kalebheitzman** [reply]
PERFECT. Thanks so much!
